worlds/stardew_valley/logic/shipping_logic.py

```diff
@@ -29,7 +28,7 @@ def can_use_shipping_bin(self) -> StardewRule:
 
     @cache_self1
     def can_ship(self, item: str) -> StardewRule:
-        return self.logic.received(Event.can_ship_items) & self.logic.has(item)
+        return self.logic.shipping.can_use_shipping_bin & self.logic.has(item)

```

Comment:
`self.can_use_shipping_bin`?

Reply:
When I designed the logic mixins, I had in mind that content packs from mods could override existing logic mixins to add more conditions to existing rules. I was specifically searching for a way to decouple the magic mod from the vanilla combat rules, because they are currently thightly coupled right now. This mean that the actual mixin used to create the `Logic` class might not be the original one, and the rules in it might be modified versions. So I think we should always use `self.logic.whatever` when creating a rule, even if it's theoritically in the same file. 

That design is documented in `worlds/stardew_valley/logic/logic_and_mods_design.md`.

Reply:
Good to know. Then my comment will be that there are other `self.method_call()` in the few methods below this one